packages/clerk-js/src/ui.retheme/elements/SocialButtons.tsx

```diff
@@ -63,112 +113,165 @@ export const SocialButtons = React.memo((props: SocialButtonsRootProps) => {
+
+            // When strategies break into 2 rows or more, use the first item of the first
+            // row as reference for the width of the buttons in the second row and beyond
+            const ref =
+              strategies.length > MAX_STRATEGIES_PER_ROW && rowIndex === 0 && strategyIndex === 0
+                ? firstStrategyRef
+                : null;
+
+            return (
+              <div key={strategy}>
+                <ButtonElement
+                  id={strategyToDisplayData[strategy].id}
+                  ref={ref}
+                  onClick={startOauth(strategy)}
+                  isLoading={card.loadingMetadata === strategy}
+                  isDisabled={card.isLoading}
+                  label={label}
+                  textLocalizationKey={localizedText}
+                  icon={
+                    <Image
+                      elementDescriptor={[descriptors.providerIcon, descriptors.socialButtonsProviderIcon]}
+                      elementId={descriptors.socialButtonsProviderIcon.setId(strategyToDisplayData[strategy].id)}
+                      isLoading={card.loadingMetadata === strategy}
+                      isDisabled={card.isLoading}
+                      src={strategyToDisplayData[strategy].iconUrl}
+                      alt={`Sign in with ${strategyToDisplayData[strategy].name}`}
+                      sx={theme => ({ width: theme.sizes.$4, height: 'auto', maxWidth: '100%' })}
+                    />
+                  }
+                />
+              </div>
+            );
+          })}
+        </Grid>
+      ))}
+    </Flex>
   );
-};
+});
 
 type SocialButtonProps = PropsOfComponent<typeof Button> & {
   icon: React.ReactElement;
   id: OAuthProvider | Web3Provider;
-  providerName: string;
+  textLocalizationKey: LocalizationKey | undefined;
   label?: string;
 };
 
-const SocialButtonIcon = (props: SocialButtonProps): JSX.Element => {
-  const { icon, label, id, providerName, ...rest } = props;
+const SocialButtonIcon = forwardRef((props: SocialButtonProps, ref: Ref<HTMLButtonElement> | null): JSX.Element => {
+  const { icon, label, id, textLocalizationKey, ...rest } = props;
+
   return (
     <Button
+      ref={ref}
       elementDescriptor={descriptors.socialButtonsIconButton}
       elementId={descriptors.socialButtonsIconButton.setId(id)}
       variant='icon'
       colorScheme='neutral'
-      sx={t => ({
-        padding: 0,
-        borderColor: t.colors.$blackAlpha200,
-      })}
+      sx={{
+        width: '100%',
+      }}
       {...rest}
     >
       {icon}
     </Button>
   );
-};
+});
 
 const SocialButtonBlock = (props: SocialButtonProps): JSX.Element => {
-  const { label, id, providerName, sx, icon, ...rest } = props;
+  const { icon, isLoading, label, textLocalizationKey, ...rest } = props;
+  const isIconElement = isValidElement(icon);
 
   return (
-    <ArrowBlockButton
-      elementDescriptor={descriptors.socialButtonsBlockButton}
```

Comment:
@marcelscruz Is this `elementDescriptor` getting removed? If so, we should keep track of these changes, because they're affecting our Public API.

Reply:
Another good catch, thanks!
I just added the descriptors and ids on the button and the text. 
On a second note, even though the formerly used `ArrowBlockButton` accepts descriptors and ids for the spinner and the icon, `SocialButton` was never passing these values to it. Should we add it?

Reply:
@marcelscruz Yes, sounds minor, but let's keep it consistent! good catch!

Reply:
The icon already had descriptors, and now all spinners also have it.